pkg/runtime/scheme.go

```diff
@@ -322,13 +322,14 @@ func (s *Scheme) Convert(in, out interface{}) error {
 // Converts the given field label and value for an apiResource field selector from
 // versioned representation to an unversioned one.
 func (s *Scheme) ConvertFieldLabel(version, apiResource, label, value string) (string, string, error) {
-	if typeFuncMap, ok := s.fieldLabelConversionFuncs[version]; ok {
-		if conversionFunc, ok := typeFuncMap[apiResource]; ok {
-			return conversionFunc(label, value)
-		}
+	if s.fieldLabelConversionFuncs[version] == nil {
+		return "", "", fmt.Errorf("No conversion function found for version: %s", version)
```

Comment:
I recommend not forcing people to write conversion function if it's unnecessary.